contracts/actions/checkers/LiquityRatioIncreaseCheck.sol

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity =0.8.10;
+
+import "../ActionBase.sol";
+import "../liquity/helpers/LiquityRatioHelper.sol";
+import "../../utils/TransientStorage.sol";
+
+contract LiquityRatioIncreaseCheck is ActionBase, LiquityRatioHelper {
+
+    /// @dev 5% offset acceptable
+    uint256 internal constant RATIO_OFFSET = 50000000000000000;
+
+    TransientStorage public constant tempStorage = TransientStorage(TRANSIENT_STORAGE);
+
+    error BadAfterRatio(uint256 startRatio, uint256 currRatio);
+
+    struct Params {
+        uint256 targetRatioIncrease;
+    }
+
+    /// @inheritdoc ActionBase
+    function executeAction(
+        bytes memory _callData,
+        bytes32[] memory _subData,
+        uint8[] memory _paramMapping,
+        bytes32[] memory _returnValues
+    ) public payable virtual override returns (bytes32) {
+        Params memory inputData = parseInputs(_callData);
+
+        uint256 targetRatioIncrease = _parseParamUint(uint256(inputData.targetRatioIncrease), _paramMapping[0], _subData, _returnValues);
+
+        address troveOwner = address(this);
+
+        (uint256 currRatio,) = getRatio(troveOwner);
+
+        uint256 targetRatio = currRatio + targetRatioIncrease;
+
+        uint256 startRatio = uint256(tempStorage.getBytes32("LIQUITY_RATIO"));
+        
+        // if repay ratio should be better off
+        if (currRatio <= startRatio) {
+            revert BadAfterRatio(startRatio, currRatio);
+        }
+
```

Comment:
we should add one more check, currRatio > (targetRatio - ratio offset)

Without it, it's allowed to raise ratio by 0.01% and the tx will go through

Reply:
startRatio = 175%
user subbed targetRatioIncrease = 25%

Without the check anything between 175 and 205 is allowed (better to raise so it's only possible to go inbetween 195 and 205) I'd even go further and lower the ratio offset